ambry-rest/src/main/java/com.github.ambry.rest/NettyServer.java

```diff
@@ -49,27 +50,20 @@
   private final NettyMetrics nettyMetrics;
   private final EventLoopGroup bossGroup;
   private final EventLoopGroup workerGroup;
-  private final RestRequestHandler requestHandler;
-  private final PublicAccessLogger publicAccessLogger;
-  private final RestServerState restServerState;
+  private final ArrayList<Map.Entry<String, ChannelHandler>> channelHandlers;
```

Comment:
Though in the current usage there is no strict key-value relationship, is there? There is an association between these entries, sure, but it is not strictly key-value as nothing stops you from having multiple keys with the same value in this `ArrayList`. That said, would a `Map` itself work here? Why not use that?